extensions/warp-ipfs/src/store/community.rs

```diff
@@ -0,0 +1,389 @@
+use super::{conversation::message::MessageDocument, topics::ConversationTopic, PeerIdExt};
+use crate::store::DidExt;
+use chrono::{DateTime, Utc};
+use core::hash::Hash;
+use indexmap::{IndexMap, IndexSet};
+use ipld_core::cid::Cid;
+use rust_ipfs::{Ipfs, Keypair};
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+use warp::{
+    crypto::DID,
+    error::Error,
+    raygun::community::{
+        Community, CommunityChannel, CommunityChannelPermission, CommunityChannelPermissions,
+        CommunityChannelType, CommunityInvite, CommunityPermission, CommunityPermissions,
+        CommunityRole, RoleId,
+    },
+};
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub struct CommunityRoleDocument {
+    pub id: RoleId,
+    pub name: String,
+    pub members: IndexSet<DID>,
+}
+impl CommunityRoleDocument {
+    pub fn new(name: String) -> Self {
+        Self {
+            id: Uuid::new_v4(),
+            name,
+            members: IndexSet::new(),
+        }
+    }
+}
+impl From<CommunityRoleDocument> for CommunityRole {
+    fn from(value: CommunityRoleDocument) -> Self {
+        let mut role = CommunityRole::default();
+        role.set_id(value.id);
+        role.set_name(value.name);
+        role.set_members(value.members);
+        role
+    }
+}
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub struct CommunityInviteDocument {
+    pub id: Uuid,
+    pub target_user: Option<DID>,
+    pub created: DateTime<Utc>,
+    pub expiry: Option<DateTime<Utc>>,
+}
+impl CommunityInviteDocument {
+    pub fn new(target_user: Option<DID>, expiry: Option<DateTime<Utc>>) -> Self {
+        Self {
+            id: Uuid::new_v4(),
+            target_user,
+            created: Utc::now(),
+            expiry,
+        }
+    }
+}
+impl From<CommunityInviteDocument> for CommunityInvite {
+    fn from(value: CommunityInviteDocument) -> Self {
+        let mut community_invite = CommunityInvite::default();
+        community_invite.set_id(value.id);
+        community_invite.set_target_user(value.target_user);
+        community_invite.set_created(value.created);
+        community_invite.set_expiry(value.expiry);
+        community_invite
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Eq)]
+pub struct CommunityDocument {
+    pub id: Uuid,
+    pub name: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub description: Option<String>,
+    pub creator: DID,
```

Comment:
we can probably rename this to `owner` since it may be possible to change ownership in the future.